srcjs/input_binding_slider.js

```diff
@@ -90,13 +122,26 @@ $.extend(sliderInputBinding, textInputBinding, {
         msg.from = data.value;
       }
     }
-    if (data.hasOwnProperty('min'))  msg.min   = data.min;
-    if (data.hasOwnProperty('max'))  msg.max   = data.max;
-    if (data.hasOwnProperty('step')) msg.step  = data.step;
+    var sliderFeatures = ['min', 'max', 'step'];
+    for (var feats of sliderFeatures) {
+      if (data.hasOwnProperty(feats)) msg[feats] = data[feats];
+    }
 
     if (data.hasOwnProperty('label'))
       $el.parent().find('label[for="' + $escape(el.id) + '"]').text(data.label);
 
+    var domElements = ['data-type', 'time-format', 'timezone'];
+    for (var elm of domElements) {
+      if (data.hasOwnProperty(elm))
+        $el.data(elm, data[elm]);
+    }
+
+    var dataType = $el.data('data-type');
```

Comment:
Should check other `receiveMessage` implementations to see if they use `$el.data()`.

Reply:
No they don't. Based on a keyword search no other `receiveMessage`s use `$el.data()`.

Reply:
We'll need to hang this state on the object somewhere, and I think maybe this is the best we can do. I can't think of any ill effects from doing this, so maybe we should just go with it.

Reply:
They is almost a way we can do it. Take a look this line: https://github.com/tmastny/shiny/blob/881fe0cfcea89c90d20b4c100fd4e16ed983286a/R/update-input.R#L431-L434

On `updateSliderInput` we recalculate `data-type` and `time-format` before sending the data to JS. The only state we have to maintain is the timezone.

The problem with the timezone is that we don't calculate it based on the POSIXl/ct, but with the `timezone` argument.